src/Engine/EmitMSIL/Replication.cs

```diff
@@ -315,14 +351,121 @@ private static List<List<ReplicationGuide>> ConstructRepGuides(string[][] replic
             return compliantTarget;
         }
 
+        private static CLRFunctionEndPoint SelectFEPFromMultiple(
```

Comment:
for these really big functions ... basically all the functions in this file, I really would not mind a summary :) - even if the methods are internal, trying to remember what they all do is not trivial for me.